numba_typing/overload_list.py

```diff
@@ -0,0 +1,183 @@
+import numpy
+import numba
+from numba import types
+from numba import typeof
+from numba.extending import overload
+from type_annotations import product_annotations, get_func_annotations
+from numba import njit
+import typing
+from numba import NumbaDeprecationWarning, NumbaPendingDeprecationWarning
+import warnings
+from numba.typed import List, Dict
+from inspect import getfullargspec
```

Comment:
There are some unused imports here, like `numpy`, `typeof` and `njit`, please clean them up.